tests/filecheck/backend/riscv/convert_ptr_to_riscv.mlir

```diff
@@ -0,0 +1,19 @@
+// RUN: xdsl-opt %s -p convert-ptr-to-riscv --split-input-file | filecheck %s
+
+%m, %idx, %v = "test.op"() : () -> (memref<3x2xi32>, index, i32)
+
+%p = ptr_xdsl.to_ptr %m : memref<3x2xi32> -> !ptr_xdsl.ptr
+// CHECK: %p = builtin.unrealized_conversion_cast %m : memref<3x2xi32> to index
+
+%r0 = ptr_xdsl.ptradd %p, %idx : (!ptr_xdsl.ptr, index) -> !ptr_xdsl.ptr
+// CHECK-NEXT:  %r0 = arith.addi %p, %idx : index
```

Comment:
This should lower to casts and riscv add assembly op

Reply:
I would say no, that it's ok to repeat in this way

Reply:
the way not to repeat in this case would be to split the pass into two, one that lowers to arith, the other to riscv, but I don't think this is worth it in this case